subsys/partition_manager/pm.yml.memfault

```diff
@@ -0,0 +1,5 @@
+#include <autoconf.h>
+
+memfault_storage:
+  placement: {before: [end]}
```

Comment:
Are there any alignment requirements for this partition? I guess it should be at least page aligned since you are doing page-erases inside the module. 

You could create a helper kconfig without prompt to expose the correct value. Here is an example on how to get the erase size https://github.com/nrfconnect/sdk-zephyr/blob/master/soc/arm/nordic_nrf/Kconfig#L29